src/icrar/leap-accelerate/algorithm/casa/PhaseRotate.cc

```diff
@@ -154,53 +153,48 @@ namespace casalib
+                casalib::RotateVisibilities(integration.get(), metadata, direction);
                 output_integrations.emplace(integration.get().integration_number, direction, boost::none);
             }
             else
             {
                 if(!metadata.avg_data.is_initialized())
                 {
-                    throw icrar::exception("avg_data must be initialized", __FILE__, __LINE__);
+                    throw exception("avg_data must be initialized", __FILE__, __LINE__);
                 }
 
-
                 casacore::Matrix<Radians> phaseAngles = casa_matrix_map(metadata.avg_data.get(), [](std::complex<double> c)
                 {
                     return std::arg(c);
                 });
 
                 auto e_phaseAngles = ToMatrix(phaseAngles);
-                Eigen::VectorXd e_phaseAnglesI1 = icrar::cpu::VectorRangeSelect(e_phaseAngles, ToVector(metadata.I1), 0); // 1st pol only
+                Eigen::VectorXd e_phaseAnglesI1 = cpu::VectorRangeSelect(e_phaseAngles, ToVector(metadata.I1), 0); // 1st pol only
                 // Value at last index of phaseAnglesI1 must be 0 (which is the reference antenna phase value)
                 e_phaseAnglesI1.conservativeResize(e_phaseAnglesI1.rows() + 1);
                 e_phaseAnglesI1(e_phaseAnglesI1.size() - 1) = 0.0;
                 auto phaseAnglesI1 = ConvertVector(e_phaseAnglesI1);
                 
-                if(metadata.Ad1.shape()[1] != phaseAnglesI1.shape()[0])
-                {
-                    std::cout << "Ad1" << metadata.Ad1.shape()[1] << "!=" << phaseAnglesI1.shape()[0] << std::endl;
-                }
-                casacore::Matrix<double> cal1 = icrar::casalib::multiply(metadata.Ad1, phaseAnglesI1);
+                casacore::Matrix<double> cal1 = casalib::multiply(metadata.Ad1, phaseAnglesI1);
 
-                Eigen::MatrixXd e_phaseAnglesI = icrar::cpu::MatrixRangeSelect(e_phaseAngles, ToVector(metadata.I), Eigen::all);
+                Eigen::MatrixXd e_phaseAnglesI = cpu::MatrixRangeSelect(e_phaseAngles, ToVector(metadata.I), Eigen::all);
                 casacore::Matrix<double> phaseAnglesI = ConvertMatrix(e_phaseAnglesI);
 
                 // Calculate DInt
                 casacore::Matrix<double> dInt = casacore::Matrix<double>(metadata.I.size() + 1, phaseAngles.shape()[1]);
                 dInt = 0;
+                constexpr double two_pi = boost::math::constants::pi<double>();
                 for(size_t n = 0; n < metadata.I.size(); ++n)
                 {
-                    dInt.row(n) = phaseAnglesI.row(n) - (casacore::sum((casacore::Array<double>)metadata.A.row(n) * (casacore::Array<double>)cal1.column(0)));
+                    double sum = casacore::sum((casacore::Array<double>)metadata.A.row(n) * (casacore::Array<double>)cal1.column(0));
+                    std::complex<double> scalear = std::exp(-sum*two_pi) + 0i;
```

Comment:
`std::complex<double> scalar {std::exp(-sum * two_pi), 0};`, the "double + complex literal" -> "complex" conversion is a bit confusing.